### PR TITLE
[MCC-177314] Integrate Rave deployment scripts into the workflow

### PR DESCRIPTION
**Pull request by junshao**
- [x] Added a hard-coded env var injecting function.
- [x] Added Rave deployment scripts into the workflow.
- [x] Removed `$replacewritehost` because Rave deployment script never uses `write-host`.

@echen-mdsol @vcapers-mdsol Please review and merge if OK. Thanks.

### CODE REVIEW

**Comment by echen-mdsol** [comment]
This doesn't work
```
        Invoke-RemoteScriptInParallel -sessions $sessions -script {
            # 0. Prepare node to run the Rave deployment scripts
            & $replacewritehost
            & $setupSession
            & $injectEnvironmentVariables
        }
```

You have to call Invoke-RemotScriptInParallel on each block individually.
```
        Invoke-RemoteScriptInParallel -sessions $sessions -script $setupSession
        Invoke-RemoteScriptInParallel -sessions $sessions -script $replacewritehost
        Invoke-RemoteScriptInParallel -sessions $sessions -script $injectEnvironmentVariables
```

**Comment by junshao** [comment]
@echen-mdsol Fixed. Thanks

**Comment by junshao** [comment]
Adding @art2003

File: conductor.ps1 [hunk 1]
```diff
@@ -61,19 +46,62 @@ function Get-NodeNames {
     return $nodes
 }
 
+$injectEnvironmentVariables = {
+    # Set all environment variables based on the input JSON string
+    $env:SITE_NAME="RaveProdTestSite"
+    $env:PACKAGE_DIR="\\hdcsharedmachine\C$\packages\Rave\2015.2.0"
```
Comment (art2003) [inline conductor.ps1]: Should we pass it as a parameter to the script

File: conductor.ps1 [hunk 2]
```diff
@@ -61,19 +46,62 @@ function Get-NodeNames {
...
+    $env:DEPLOY_ID ="yyyyMMddhhmmss-buildid"
+    $env:RELEASE_DIR="C:\MedidataApp\Rave\Sites\$env:SITE_NAME\release\$env:DEPLOY_ID"
+    $env:ARTIFACTS_DIR="C:\MedidataApp\Rave\Sites\$env:SITE_NAME\artifacts\$env:DEPLOY_ID"
+}
+
+$installDeploymentScripts = {
+    $packagePath = "$env:PACKAGES_DIR\Medidata.AdminProcess.zip"
+    $artifactPath = "$env:ARTIFACTS_DIR\Medidata.AdminProcess.zip"
+    $releaseDir = "$env:RELEASE_DIR\Medidata.AdminProcess"
```
Comment (art2003) [inline conductor.ps1]: Should we check that packagePath and artifactPath are exist

**Comment by echen-mdsol** [comment]
I added `$replacewritehost` to capture any host messages from the remote nodes. The deploy tasks use Write-Host to log messages all over the place. I figured we would want to capture these messages in the log file.

Powershell doesn't let you to capture host messages by default. You can't even forward them to the output stream. It's a bit hacky but it was the only way I could find around Powershell's limitation.

File: conductor.ps1 (hunk 2 above)
Comment (junshao) [inline conductor.ps1]: Will add. Thanks

**Comment by junshao** [comment]
Reverted the commit that removing `$replacewritehost`

**Comment by junshao** [comment]
Just discussed with @echen-mdsol , as Rave deployment script isn't using `Write-Host` at all, the `$replacewritehost` is believed not required. Will restore the commit that removing `$replacewritehost` back.

**Comment by junshao** [comment]
@echen-mdsol Added your commit for refactoring logging.

**Comment by junshao** [comment]
@art2003 @echen-mdsol I think we can merge this PR and then do improvements and fixes.

**Pull request merged**